v3/src/components/map/utilities/lat-lng-grid.ts

```diff
@@ -0,0 +1,119 @@
+import {LatLngBounds} from "leaflet"
+
+export class LatLngGridCell {
+  bounds: LatLngBounds
+  cases: string[]
+  selected: boolean
+
+  constructor(bounds: LatLngBounds) {
+    this.bounds = bounds
+    this.cases = []
+    this.selected = false
+  }
+
+  get count(): number {
+    return this.cases.length
+  }
+}
+
+export class LatLngGrid {
+  maxCount: number
+  gridCells: (LatLngGridCell | undefined)[][]
+
+  constructor() {
+    this.maxCount = 0
+    this.gridCells = []
+  }
+
+  getGridCell(iLongIndex: number, iLatIndex: number): LatLngGridCell | undefined {
+    return this.gridCells[iLongIndex]?.[iLatIndex]
+  }
+
+  setGridCell(iLongIndex: number, iLatIndex: number, latLngGridCell: LatLngGridCell): void {
+    if (!this.gridCells[iLongIndex]) {
+      this.gridCells[iLongIndex] = []
+    }
+    this.gridCells[iLongIndex][iLatIndex] = latLngGridCell
+  }
+
+  forEachGridCell(iFunc: (gridCell: LatLngGridCell, longIndex: number, latIndex: number) => void): void {
+    this.gridCells.forEach((iLongArray, iLongIndex) => {
+      iLongArray.forEach((iGridCell, iLatIndex) => {
+        iGridCell && iFunc(iGridCell, iLongIndex, iLatIndex)
+      })
+    })
+  }
+
+  setupGridFromBounds(latLongBounds: LatLngBounds | undefined,
+                      setGridSize: (iSize: number) => void, getGridSize: () => number):
+    { startWest: number, startSouth: number } {
+
+
+    const kGridCellCount = 20,
+      boundsCenter = latLongBounds?.getCenter(),
+      refLong = boundsCenter?.lng ?? 0,
+      refLat = boundsCenter?.lat ?? 0
+    let boundsWest = latLongBounds?.getWest() ?? 0,
+      boundsEast = latLongBounds?.getEast() ?? 0,
+      boundsSouth = latLongBounds?.getSouth() ?? 0,
+      boundsNorth = latLongBounds?.getNorth() ?? 0
+
+    if (boundsWest === boundsEast) {
+      boundsWest -= 5 // arbitrary offset
+      boundsEast += 5
+    }
+    if (boundsNorth === boundsSouth) {
+      boundsNorth += 5 // arbitrary offset
+      boundsSouth -= 5
+    }
+
+    setGridSize(Math.min((boundsEast - boundsWest) /
+      kGridCellCount, (boundsNorth - boundsSouth) / kGridCellCount))
```

Comment:
The trouble is that the gridSize you set is not the gridSize you get back because of the use of _gridMultiplier. I elected not to pass that in, opting for functions instead.